pkg/lib/flavor/util/platform_flavor_util.go

```diff
@@ -210,33 +211,38 @@ func (pfutil PlatformFlavorUtil) GetHardwareSectionDetails(hostInfo *taModel.Hos
 		// Set TPM Feature presence
 		tpm := cm.TPM{}
 		tpm.Enabled = hostInfo.HardwareFeatures.TPM.Enabled
-		tpm.Enabled = hostInfo.HardwareFeatures.TPM.Enabled
-		tpm.Version = hostInfo.HardwareFeatures.TPM.Meta.TPMVersion
+		tpm.Meta.TPMVersion = hostInfo.HardwareFeatures.TPM.Meta.TPMVersion
 		// split into list
-		tpm.PcrBanks = strings.Split(hostInfo.HardwareFeatures.TPM.Meta.PCRBanks, constants.PCRBankSeparator)
-		feature.TPM = &tpm
-
-		txt := cm.TXT{}
-		if hostInfo.HardwareFeatures.TXT != nil {
-			// Set TXT Feature presence
-			txt.Enabled = hostInfo.HardwareFeatures.TXT.Enabled
-			feature.TXT = &txt
-		}
+		tpm.Meta.PCRBanks = strings.Split(hostInfo.HardwareFeatures.TPM.Meta.PCRBanks, constants.PCRBankSeparator)
+		feature.TPM = tpm
+
+		txt := cm.HardwareFeature{}
+		// Set TXT Feature presence
+		txt.Enabled = hostInfo.HardwareFeatures.TXT.Enabled
+		feature.TXT = txt
 
 		cbnt := cm.CBNT{}
 		// set CBNT
-		if hostInfo.HardwareFeatures.CBNT != nil {
-			cbnt.Enabled = hostInfo.HardwareFeatures.CBNT.Enabled
-			cbnt.Profile = hostInfo.HardwareFeatures.CBNT.Meta.Profile
-			feature.CBNT = &cbnt
-		}
-
-		suefi := cm.SUEFI{}
-		// and SUEFI state
-		if hostInfo.HardwareFeatures.SUEFI != nil {
-			suefi.Enabled = hostInfo.HardwareFeatures.SUEFI.Enabled
-			feature.SUEFI = &suefi
-		}
+		cbnt.Enabled = hostInfo.HardwareFeatures.CBNT.Enabled
+		cbnt.Meta.Profile = hostInfo.HardwareFeatures.CBNT.Meta.Profile
+		cbnt.Meta.MSR = hostInfo.HardwareFeatures.CBNT.Meta.MSR
+		feature.CBNT = cbnt
```

Comment:
I think hardware feature information needs to be populated post validation if they are present

Reply:
As per new design, all the details will be provided in platform info. Followed the same for host manifest and flavor. Will that be okay ?

Reply:
yes, but what if platform does not support it?

Reply:
The key will be present but it will be populated as false.

Reply:
That's not the right behavior